global-script.lua

```diff
@@ -3088,4 +3148,87 @@ function tCompare(t1,t2)
     return table.concat(newTab,"|")
     end
     if cc2(t1) == cc2(t2) then return true else return false end
-end
\ No newline at end of file
+end
+
+
+function swapPlayerAreas(a, b)
+    local function tableSwap(table)
+        local temp = table[a]
+        table[a] = table[b]
+        table[b] = temp
+    end
+    local function tintSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        if type(oa) == "string" then
+            oa = getObjectFromGUID(oa)
+            ob = getObjectFromGUID(ob)
+        end
+        local ta = oa.getColorTint()
+        local tb = ob.getColorTint()
+        oa.setColorTint(tb)
+        ob.setColorTint(ta)
+        tableSwap(table)
+    end
+    local function positionSwap(table)
+        local oa = table[a]
+        local ob = table[b]
+        if type(oa) == "string" then
+            oa = getObjectFromGUID(oa)
+            ob = getObjectFromGUID(ob)
+        end
+        local ta = oa.getPosition()
+        local tb = ob.getPosition()
+        oa.setPosition(tb)
+        ob.setPosition(ta)
+    end
+
+    print("Player " .. a .. " is trading places with player " .. b .. ".")
+    for i = 1,2 do
+        local ta = Player[a].getHandTransform(i)
+        local tb = Player[b].getHandTransform(i)
+        Player[a].setHandTransform(tb, i)
+        Player[b].setHandTransform(ta, i)
+    end
+    tintSwap(playerTables)
+    tableSwap(playerBlocks)
+    positionSwap(defendBags)
+    positionSwap(readyTokens)
+    tableSwap(elementScanZones)
+    if playerBlocks[a] then playerBlocks[a].setVar("playerColor", a) end
+    if playerBlocks[b] then playerBlocks[b].setVar("playerColor", b) end
+end
+
+
+-- What to do when the swap player colors button is clicked.
+function onSwapButtonClicked(target_obj, source_color, alt_click)
+    player = Player[source_color]
+    target_color = target_obj.getVar("playerColor")
+    source_obj = playerBlocks[source_color]
+
+    if not target_color then
+        player.print("That seat is already taken.")
```

Comment:
why use `print` instead of `broadcastToColor`?

Reply:
I didn't know it existed!  Changing to use that.